### ISSUE TITLE
macro BUFFER_STDARG

### ISSUE THREAD

**Issue by ghost** [faelys/libsoldout#26]
What's the point of the macro [BUFFER_STDARG](https://github.com/faelys/libsoldout/blob/bafd72d3d0893f4d57c5f418a38ec8215ebf17cf/buffer.c#L19) ?

inside `#ifdef BUFFER_STDARG` must be `bufprintf()` function declaration as it depends on `vbufprintf()` function in [buffer.h](https://github.com/faelys/libsoldout/blob/e458e333f4a1937e428879fedd09b3439481216e/buffer.h#L153-L160);
+ there is no  `#ifdef BUFFER_STDARG` for those functions in `buffer.c`..

So, should it be fixed, or could it be safe removed ?

**Comment by faelys** [comment]
That's an interesting question…

I think it can be safely removed.

The point is a bit hazy, as you might have seen from the repository history, this macro is older than the project, and my memory isn't perfect. I remember there was some issue in some compilation unit that used the buffer but couldn't include `stdarg.h`. Maybe it had something to do with Windows build, it was around that time that I tried (without much success) to cross-build some UI for Windows on a linux host, and Microsoft's stdarg was special by then. Maybe it was just my sheer incompetence :-)

Anyway, nowadays even Windows has a standard-compliant `stdarg.h` with `va_copy` (from I can read on msdn), so I don't see the point of keeping the macro.

On a slight tangent but staying on the historical mood, `BUFFER_STATS` was used a long time ago to track a subtle memory leak (a few hundreds of bytes per month), and was kept as a metric to monitor on my personal webserver (that was replaced months ago). However unlike `BUFFER_STDARG`, it might have some value to some users even if I don't need it anymore myself.
